smex/elf.c

```diff
@@ -451,17 +451,25 @@ int elf_read_section(const struct elf_module *module, const char *section_name,
 		return -ENOMEM;
 
 	/* fill buffer with section content */
-	fseek(module->fd, section->off, SEEK_SET);
-	read = fread(*dst_buff, 1, section->size, module->fd);
-	if (read != section->size) {
-		fprintf(stderr,
-			"error: can't read %s section %d\n", section_name,
-			-errno);
-		free(*dst_buff);
-		return -errno;
+	ret = fseek(module->fd, section->off, SEEK_SET);
+	if (ret) {
+		fprintf(stderr, "error: can't seek to %s section %d\n", section_name, -errno);
+		ret = -errno;
+		goto error;
+	}
+
+	ret = fread(*dst_buff, 1, section->size, module->fd);
+	if (ret != section->size) {
+		fprintf(stderr, "error: can't read %s section %d\n", section_name, -errno);
+		ret = -ENODATA;
```

Comment:
I'd do something like `ret = ret < 0 ? -errno : -ENODATA;` but at least this doesn't miss errors